paasta_tools/tron_tools.py

```diff
@@ -453,6 +419,47 @@ def create_complete_config(service, soa_dir=DEFAULT_SOA_DIR):
     )
 
 
+def validate_complete_config(service: str, cluster: str, soa_dir: str=DEFAULT_SOA_DIR) -> List[str]:
+    job_configs, other_config = load_tron_service_config(service, cluster, soa_dir)
+
+    if service != MASTER_NAMESPACE and other_config:
+        other_keys = list(other_config.keys())
+        return [
+            f'Non-{MASTER_NAMESPACE} namespace cannot have other config values, found {other_keys}',
+        ]
+
+    # PaaSTA-specific validation
+    for job_config in job_configs:
+        check_msgs = job_config.validate()
+        if check_msgs:
+            return check_msgs
+
+    # Use Tronfig on generated config from PaaSTA to validate the rest
+    other_config['jobs'] = [
+        format_tron_job_dict(
+            job_config=job_config,
+            cluster_fqdn_format='{cluster}',
+            default_paasta_cluster=None,
+        ) for job_config in job_configs
+    ]
+    complete_config = yaml.dump(other_config, Dumper=Dumper)
+
+    proc = subprocess.run(
+        ['tronfig', '-', '-V', '-n', service],
+        input=complete_config,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        encoding='utf-8',
+    )
+    if proc.returncode != 0:
+        process_errors = proc.stderr.strip()
+        if process_errors:  # There was an error in validation
+            print(proc.stderr)
```

Comment:
Using paasta_print here? Or could we combine proc.stderr and proc.stdout and return them together?

Reply:
I'll change it to paasta_print. The reason I don't return stderr is because I don't want validation to fail if the `tronfig` command fails to run for some reason (e.g. it doesn't exist). If `tronfig` runs, whether or not validation passed, all the messages will be in stdout.